src/test/clip.t.sol

```diff
@@ -692,4 +692,85 @@ contract ClipperTest is DSTest {
         uint256 lotReturn = 30 ether - expectedGem;         // lot - loaf.tab / max = 15
         assertEq(vat.gem(ilk, me), 960 ether + lotReturn);  // Collateral returned (10 WAD)
     } 
+
+    function auctionResetSetup(uint256 tau) internal {
+        LinearDecrease calc = new LinearDecrease();
+        calc.file(bytes32("tau"), tau);   // 10 hours till zero is reached (used to test tail)  
+
+        clip.file("buf",  ray(1.25 ether)); // 25% Initial price buffer
+        clip.file("dust", rad(20   ether)); // $20 dust
+        clip.file("calc", address(calc));   // File price contract
+        clip.file("cusp", ray(0.5 ether));  // 50% drop before reset
+        clip.file("tail", 3600);            // 1 hour before reset
+
+        assertEq(clip.kicks(), 0);
+        dog.bark(ilk, me);
+        assertEq(clip.kicks(), 1);
+    }
+
+    function testFail_auction_reset_tail() public {
```

Comment:
This test is correct, however it doesn't check exactly the same than the next one.
I mean I could expect the `pip.poke` had something to do with the next test passing, and not the extra second in the `hevm.warp`.
In general the `testFail` is not very informative. We might want to start checking failures everywhere wrapping everything in `try_call` kind functions and incorporating them in the same test, like:
```
hevm.warp(startTime + 3600 seconds);
assertEq(!try_redo(1));
hevm.warp(startTime + 3601 seconds);
assertEq(try_redo(1));
```

Anyway if we want to convert all the tests to that way we could do a new PR.

Reply:
This is a good point...every time I write a `testFail`, I have to comment out the line that's supposed to revert and make sure it fails without that line...and some change that makes an earlier line revert can hide a regression in the behavior of the later call. It's just a fragile pattern.

I think we just discovered some technical debt :) (since we use `testFail` in more places than here).